### PR TITLE
[CI]enable AMP accuracy test for inductor on SPR CPU

### PR DESCRIPTION
**Pull request by DiweiSun**
This is to enable inductor AMP accuracy check for on CPU in CI workflow to capture issue early. Three suites are included: timms, huggingface as well as torchbench. 

cc @mrshenli @pritamdamania87 @zhaojuanmao @satgera @gqchen @aazzolini @osalpekar @jiayisuse @H-Huang @kwen2501 @awgu @penguinwu @fegin @XilunWu @wanchaol @fduwjj @wz337 @tianyu-l @wconstab @yf225 @chauhang @d4l3k @jgong5 @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10 @gujinghui @PenghuiCheng @jianyuh @min-jean-cho @yanbing-j @Guobing-Chen @Xia-Weiwen @snadampal @mcarilli @ptrblck @leslie-fang-intel @EikanWang @voznesenskym @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @desertfire @LucasLLC

### CODE REVIEW

File: .github/workflows/inductor.yml [hunk 1]
```diff
@@ -120,11 +120,21 @@ jobs:
...
+          { config: "cpu_inductor_timm_amp", shard: 1, num_shards: 2, runner: "linux.c7i.metal-24xl" },
+          { config: "cpu_inductor_timm_amp", shard: 2, num_shards: 2, runner: "linux.c7i.metal-24xl" },
+          { config: "cpu_inductor_torchbench_amp", shard: 1, num_shards: 2, runner: "linux.c7i.metal-24xl" },
+          { config: "cpu_inductor_torchbenc_amp", shard: 2, num_shards: 2, runner: "linux.c7i.metal-24xl" },
           { config: "dynamic_cpu_inductor_huggingface", shard: 1, num_shards: 1, runner: "linux.12xlarge" },
           { config: "dynamic_cpu_inductor_timm", shard: 1, num_shards: 2, runner: "linux.12xlarge" },
           { config: "dynamic_cpu_inductor_timm", shard: 2, num_shards: 2, runner: "linux.12xlarge" },
           { config: "dynamic_cpu_inductor_torchbench", shard: 1, num_shards: 2, runner: "linux.12xlarge" },
           { config: "dynamic_cpu_inductor_torchbench", shard: 2, num_shards: 2, runner: "linux.12xlarge" },
+          { config: "dynamic_cpu_inductor_huggingface_amp", shard: 1, num_shards: 1, runner: "linux.c7i.metal-24xl" },
```
Comment (chuanqi129) [inline .github/workflows/inductor.yml:133]: Let's add amp static shape as the first step

File: .github/workflows/inductor.yml [hunk 2]
```diff
@@ -120,11 +120,21 @@ jobs:
           { config: "cpu_inductor_timm", shard: 2, num_shards: 2, runner: "linux.12xlarge" },
           { config: "cpu_inductor_torchbench", shard: 1, num_shards: 2, runner: "linux.12xlarge" },
           { config: "cpu_inductor_torchbench", shard: 2, num_shards: 2, runner: "linux.12xlarge" },
+          { config: "cpu_inductor_huggingface_amp", shard: 1, num_shards: 1, runner: "linux.c7i.metal-24xl" },
+          { config: "cpu_inductor_timm_amp", shard: 1, num_shards: 2, runner: "linux.c7i.metal-24xl" },
+          { config: "cpu_inductor_timm_amp", shard: 2, num_shards: 2, runner: "linux.c7i.metal-24xl" },
+          { config: "cpu_inductor_torchbench_amp", shard: 1, num_shards: 2, runner: "linux.c7i.metal-24xl" },
+          { config: "cpu_inductor_torchbenc_amp", shard: 2, num_shards: 2, runner: "linux.c7i.metal-24xl" },
```
Comment (chuanqi129) [inline .github/workflows/inductor.yml:127]:
```suggestion
          { config: "cpu_inductor_huggingface_amp", shard: 1, num_shards: 1, runner: "linux.16xlarge.spr" },
          { config: "cpu_inductor_timm_amp", shard: 1, num_shards: 2, runner: "linux.16xlarge.spr" },
          { config: "cpu_inductor_timm_amp", shard: 2, num_shards: 2, runner: "linux.16xlarge.spr" },
          { config: "cpu_inductor_torchbench_amp", shard: 1, num_shards: 2, runner: "linux.16xlarge.spr" },
          { config: "cpu_inductor_torchbenc_amp", shard: 2, num_shards: 2, runner: "linux.16xlarge.spr" },
```

Comment (chuanqi129) [reply .github/workflows/inductor.yml:127]: runner label refer to https://github.com/pytorch/test-infra/blob/main/.github/scale-config.yml#L36

File: .ci/pytorch/test.sh [hunk 3]
```diff
@@ -542,7 +542,11 @@ test_dynamo_benchmark() {
     test_single_dynamo_benchmark "dashboard" "$suite" "$shard_id" "$@"
   else
     if [[ "${TEST_CONFIG}" == *cpu_inductor* ]]; then
-      test_single_dynamo_benchmark "inference" "$suite" "$shard_id" --inference --float32 "$@"
+      if [[ "${TEST_CONFIG}" == *amp* ]]; then
+        test_single_dynamo_benchmark "inference" "$suite" "$shard_id" --inference --bfloat16 "$@"
+      else
+        test_single_dynamo_benchmark "inference" "$suite" "$shard_id" --inference --float32 "$@"
+      fi
```
Comment (chuanqi129) [inline .ci/pytorch/test.sh:549]:
```suggestion
      local dt="float32"
      if [[ "${TEST_CONFIG}" == *amp* ]]; then
        dt="amp"
      fi
      test_single_dynamo_benchmark "inference" "$suite" "$shard_id" --inference --"$dt" "$@"
```

**Review by chuanqi129 — APPROVED** [review]

**Review by jgong5 — APPROVED** [review]

**Review by desertfire — CHANGES_REQUESTED** [review]

File: .github/workflows/inductor.yml [hunk 4]
```diff
@@ -120,6 +120,11 @@ jobs:
           { config: "cpu_inductor_timm", shard: 2, num_shards: 2, runner: "linux.12xlarge" },
           { config: "cpu_inductor_torchbench", shard: 1, num_shards: 2, runner: "linux.12xlarge" },
           { config: "cpu_inductor_torchbench", shard: 2, num_shards: 2, runner: "linux.12xlarge" },
+          { config: "cpu_inductor_huggingface_amp", shard: 1, num_shards: 1, runner: "linux.16xlarge.spr" },
+          { config: "cpu_inductor_timm_amp", shard: 1, num_shards: 2, runner: "linux.16xlarge.spr" },
+          { config: "cpu_inductor_timm_amp", shard: 2, num_shards: 2, runner: "linux.16xlarge.spr" },
+          { config: "cpu_inductor_torchbench_amp", shard: 1, num_shards: 2, runner: "linux.16xlarge.spr" },
+          { config: "cpu_inductor_torchbenc_amp", shard: 2, num_shards: 2, runner: "linux.16xlarge.spr" },
```
[the remark this answers is not in the document]
Comment (DiweiSun) [reply .github/workflows/inductor.yml]: Oops, sorry for the typo. Corrected. 

Thanks!

**Comment by DiweiSun** [comment]
@jgong5 , could you please kindly help review it again? Thanks

**Review by desertfire — APPROVED** [review]

**Comment by DiweiSun** [comment]
All failures happens on NVDIA GPU, however the PR is not supposed to impact on them.

**Review by atalman — APPROVED** [review]

**Comment by lezcano** [comment]
Bad rebase. Too many people tagged. Please submit a new PR.

**Pull request closed without merging**